mouse.go

```diff
@@ -28,38 +46,216 @@ func (m MouseEvent) String() (s string) {
+		}
+	} else {
+		s += mouseEventTypes[m.Type]
+	}
+
 	return s
 }
 
+// MouseAction represents the action that occurred during a mouse event.
+type MouseAction int
+
+// Mouse event actions
+const (
+	MouseActionPress MouseAction = iota
+	MouseActionRelease
+	MouseActionMotion
+)
+
+var mouseActions = map[MouseAction]string{
+	MouseActionPress:   "press",
+	MouseActionRelease: "release",
+	MouseActionMotion:  "motion",
+}
+
+// MouseButton represents the button that was pressed during a mouse event.
+type MouseButton int
+
+// Mouse event buttons
+//
+// This is based on X11 mouse button codes.
+//
+//	1 = left button
+//	2 = middle button (pressing the scroll wheel)
+//	3 = right button
+//	4 = turn scroll wheel up
+//	5 = turn scroll wheel down
+//	6 = push scroll wheel left
+//	7 = push scroll wheel right
+//	8 = 4th button (aka browser backward button)
+//	9 = 5th button (aka browser forward button)
+//
+// Other buttons are not supported.
+const (
+	MouseButtonNone MouseButton = iota
+	MouseButtonLeft
+	MouseButtonMiddle
+	MouseButtonRight
+	MouseButtonWheelUp
+	MouseButtonWheelDown
+	MouseButtonWheelLeft
+	MouseButtonWheelRight
+	MouseButtonBackward
+	MouseButtonForward
+	MouseButton10
+	MouseButton11
+
+	MouseButtonUnknown
+)
+
+var mouseButtons = map[MouseButton]string{
+	MouseButtonNone:       "none",
+	MouseButtonLeft:       "left",
+	MouseButtonMiddle:     "middle",
+	MouseButtonRight:      "right",
+	MouseButtonWheelUp:    "wheel up",
+	MouseButtonWheelDown:  "wheel down",
+	MouseButtonWheelLeft:  "wheel left",
+	MouseButtonWheelRight: "wheel right",
+	MouseButtonBackward:   "backward",
+	MouseButtonForward:    "forward",
+	MouseButton10:         "button 10",
+	MouseButton11:         "button 11",
+	MouseButtonUnknown:    "unknown",
+}
+
 // MouseEventType indicates the type of mouse event occurring.
+//
+// Deprecated: Use MouseAction & MouseButton instead.
 type MouseEventType int
 
 // Mouse event types.
+//
+// Deprecated: Use MouseAction & MouseButton instead.
 const (
 	MouseUnknown MouseEventType = iota
 	MouseLeft
 	MouseRight
 	MouseMiddle
-	MouseRelease
+	MouseRelease // mouse button release (X10 only)
 	MouseWheelUp
 	MouseWheelDown
+	MouseWheelLeft
+	MouseWheelRight
+	MouseBackward
+	MouseForward
+	MouseBtn10
+	MouseBtn11
 	MouseMotion
 )
 
 var mouseEventTypes = map[MouseEventType]string{
-	MouseUnknown:   "unknown",
-	MouseLeft:      "left",
-	MouseRight:     "right",
-	MouseMiddle:    "middle",
-	MouseRelease:   "release",
-	MouseWheelUp:   "wheel up",
-	MouseWheelDown: "wheel down",
-	MouseMotion:    "motion",
+	MouseUnknown:    "unknown",
+	MouseLeft:       "left",
+	MouseRight:      "right",
+	MouseMiddle:     "middle",
+	MouseRelease:    "release",
+	MouseWheelUp:    "wheel up",
+	MouseWheelDown:  "wheel down",
+	MouseWheelLeft:  "wheel left",
+	MouseWheelRight: "wheel right",
+	MouseBackward:   "backward",
+	MouseForward:    "forward",
+	MouseBtn10:      "button 10",
+	MouseBtn11:      "button 11",
```

Comment:
I almost wonder if we need to add these two at all since we're deprecating this API anyway.

Reply:
Good call!